### PR TITLE
[ROCm] Switch rocm build to use ROCm 3.7

### PR DESCRIPTION
**Pull request by deven-amd**
This PR has the changes required to switch the `--config=rocm` build to use ROCm 3.7

--------------------------------------

/cc @chsigg @cheshire @nvining-work

### CODE REVIEW

File: tensorflow/python/ops/parallel_for/math_test.py [hunk 1]
```diff
@@ -82,6 +82,11 @@ def test_unary_cwise_complex_ops(self):
     self._test_unary_cwise_ops(complex_ops, True)
 
   def test_unary_cwise_real_ops_1(self):
+    if test.is_built_with_rocm():
```
Comment (cheshire) [inline tensorflow/python/ops/parallel_for/math_test.py:85]: This is not a blocker for this PR, but could you

1) Ideally, add a decorator for skipping such tests similar to https://cs.opensource.google/tensorflow/tensorflow/+/master:tensorflow/python/eager/def_function_xla_jit_test.py;l=134?q=def_function_xla_jit%20file:tensorflow
2) While (1) is not done, use `self.skipTest` instead?

**Review by cheshire — CHANGES_REQUESTED** [review]

File: tensorflow/python/ops/parallel_for/math_test.py (hunk 1 above)
Comment (deven-amd) [reply tensorflow/python/ops/parallel_for/math_test.py:85]: @cheshire 

implemented `#2` above. Will file a separate PR to add the decorator requested in `#1` and switch all possible uses of `skipTest(('Fails on ROCm...')` to it

**Comment by deven-amd** [comment]
@chsigg gentle ping

**Comment by deven-amd** [comment]
@chsigg gentle ping

**Comment by deven-amd** [comment]
@chsigg gentle ping

**Comment by deven-amd** [comment]
@chsigg gentle ping

**Comment by jessisar** [comment]
gentle ping

**Comment by deven-amd** [comment]
@chsigg gentle ping

**Review by cheshire — DISMISSED** [review]

**Review by rthadur — DISMISSED** [review]

**Comment by gbaned** [comment]
@deven-amd Can you please resolve conflicts? Thanks!

**Comment by deven-amd** [comment]
@gbaned rebased the PR to remove the merge conflict

**Comment by deven-amd** [comment]
@cheshire please re-approve..thanks

**Pull request merged**